File: App/Models/InvoiceLine.cs
using System.ComponentModel.DataAnnotations;

namespace App.Models;

public class InvoiceLine
{
    [Key] public int Id { get; set; }
    public int InvoiceId { get; set; }
    public int ProductId { get; set; }
    public int SizeId { get; set; }
    public int Quantity { get; set; }
    public double Price { get; set; }
    public double Total { get; set; }
    public Product? Product { get; set; }
    public Invoice? Invoice { get; set; }
}

File: App/ClientApp/src/app/Features/cart/models/invoice-line.ts
import { ISize } from "../../../Shared/interfaces/size";
import { Product } from "../../../Shared/models/product";

export class InvoiceLine
{
  public id: number;
  public invoiceId: number;
  public productId: number;
  public sizeId: number;
  public product: Product;
  public quantity: number;
  public size: ISize;
  public price: number;
  public total: number;

  constructor( props: Partial<InvoiceLine> = {} )
  {
    Object.assign( this, props );
  }
}
